src_c/polygon.c

```diff
@@ -315,6 +319,73 @@ pg_polygon_dealloc(pgPolygonObject *self)
     Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
+static PyObject *
+pg_polygon_normal_polygon(PyObject *_null, PyObject *const *args,
+                          Py_ssize_t nargs)
+{
+    int sides;
+    double radius;
+    double angle = 0;
+    double Cx, Cy;
+
+    if (nargs == 3) {
+        if (!PyLong_Check(args[0])) {
+            goto error;
+        }
+        sides = (int)PyLong_AsLong(args[0]);
+
+        if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
+            !pg_DoubleFromObj(args[2], &radius)) {
+            goto error;
+        }
+    }
+    else if (nargs == 4) {
+        if (!PyLong_Check(args[0])) {
+            goto error;
+        }
+        sides = (int)PyLong_AsLong(args[0]);
+
+        if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
+            !pg_DoubleFromObj(args[2], &radius) ||
+            !pg_DoubleFromObj(args[3], &angle)) {
+            goto error;
+        }
+    }
+    else {
+        goto error;
+    }
+
+    angle *= PI / 180.0;
+
+    if (sides < 3) {
+        if (sides < 0) {
+            return RAISE(PyExc_ValueError,
+                         "the sides can not be a negative number");
+        }
+        return RAISE(PyExc_ValueError, "polygons need at least 3 sides");
+    }
+
+    double *verticies = PyMem_New(double, sides * 2);
+    if (!verticies) {
+        return PyErr_NoMemory();
+    }
+
+    int loop;
+    for (loop = 0; loop < sides; loop++) {
+        verticies[loop * 2 + 0] =
+            Cx + radius * cos(angle + PI * 2 * loop / sides);
+        verticies[loop * 2 + 1] =
+            Cy + radius * sin(angle + PI * 2 * loop / sides);
+    }
+
+    PyObject *ret = pgPolygon_New2(verticies, sides);
+    PyMem_Free(verticies);
+
+    return ret;
+error:
+    return RAISE(PyExc_TypeError, "mismatched argument types");
```

Comment:
I would specify the required arguments number in this error, so:
```suggestion
    return RAISE(PyExc_TypeError, "mismatched argument types, must either be 3 or 4");
```

Reply:
This one is still up

Reply:
already handled, the wrong number of arguments is caught early in the function